packages/@aws-cdk/custom-resource-handlers/lib/custom-resources-framework/README.md

```diff
@@ -0,0 +1,87 @@
+  },
+  'aws-stepfunctions-tasks': {
+    // contains multiple custom resource provider modules
+    'eval-nodejs-provider': [
+      {
+        // the custom resource provider type - `CdkSingletonFunction`
+        type: ProviderType.CDK_SINGLETON_FUNCTION,
+        sourceCode: path.resolve(__dirname, '..', 'aws-stepfunctions-tasks', 'eval-nodejs-handler', 'index.ts'),
+        compatibleRuntimes: [Runtime.NODEJS_18_X],
+      },
+    ],
+    'role-policy-provider': [
+      {
+        type: ProviderType.CDK_SINGLETON_FUNCTION,
+        sourceCode: path.resolve(__dirname, '..', 'aws-stepfunctions-tasks', 'role-policy-handler', 'index.py'),
+        compatibleRuntimes: [Runtime.PYTHON_3_9],
+        // prevent minify and bundle flag is set since the source code is a python file
+        preventMinifyAndBundle: true,
+      },
+    ],
+  },
+};
+```
+
+Code generation for the provider modules is triggered when this package - `@aws-cdk/custom-resource-handlers` - is built. Importantly, this framework is also responsible for minifying and bundling the custom resource providers' source code and dependencies. A flag named `preventMinifyAndBundle` can be configured as part of the `ProviderProps` to prevent minifying and bundling the source code for a specific provider. This flag is `false` by default and is only needed for Python files or for JavaScript/TypeScript files containing require imports.
+
+Once built, all generated code and bundled source code will be written to `@aws-cdk/custom-resource-handlers/dist`. The top level field in the [config](./config.ts) file defining individual `aws-cdk-lib` modules will be used to create specific directories within `@aws-cdk/custom-resource-handlers/dist` and each provider module will be a separate code generated file within these directories named as `<provider-module>.generated.ts`. As an example, the sample [config](./config.ts) file above would create the following file structure:
+
+|--- @aws-cdk
+|   |--- custom-resource-handlers
+|   |   |--- dist
+|   |   |   |--- aws-s3
+|   |   |   |   |--- replica-handler
+|   |   |   |   |   |--- index.js
+|   |   |   |   |--- replica-provider.generated.ts
+|   |   |   |--- aws-stepfunctions-tasks
+|   |   |   |   |--- eval-nodejs-handler
+|   |   |   |   |   |--- index.js
+|   |   |   |   |--- role-policy-handler
+|   |   |   |   |   |--- index.py
+|   |   |   |   |--- eval-nodejs-provider.generated.ts
+|   |   |   |   |--- role-policy-provider.generated.ts
+
+The code generated custom resource providers are usable in `aws-cdk-lib` once `aws-cdk-lib` is built. The custom resource providers will be consumable from `aws-cdk-lib/custom-resource-handlers/dist` and the file structure therein will match what was generated in `@aws-cdk/custom-resource-handlers/dist` except for providers defined in `core`. To prevent circular dependencies, all custom resource providers defined in `core`and any associated source code will be consumable from `aws-cdk-lib/core/dist/core`.
```

Comment:
`aws-cdk-lib/core/dist/core` as a path is a bit confusing. Can it be something like `aws-cdk-lib/core/dist/custom-resource-handlers`